pkg/proxy/winkernel/hns_test.go

```diff
@@ -42,95 +42,65 @@ const (
 	protocol          = 6
 	internalPort      = 80
 	externalPort      = 32440
+	policyConst       = "50"
+	isolationID       = 5000
 )
 
-func TestGetNetworkByName(t *testing.T) {
-	hnsV1 := hnsV1{}
-	hnsV2 := hnsV2{}
-
-	testGetNetworkByName(t, hnsV1)
-	testGetNetworkByName(t, hnsV2)
-}
-func TestGetEndpointByID(t *testing.T) {
-	hnsV1 := hnsV1{}
-	hnsV2 := hnsV2{}
-
-	testGetEndpointByID(t, hnsV1)
-	testGetEndpointByID(t, hnsV2)
-}
-func TestGetEndpointByIpAddressAndName(t *testing.T) {
-	hnsV1 := hnsV1{}
-	hnsV2 := hnsV2{}
-
-	testGetEndpointByIpAddressAndName(t, hnsV1)
-	testGetEndpointByIpAddressAndName(t, hnsV2)
-}
-func TestCreateEndpointLocal(t *testing.T) {
-	hnsV1 := hnsV1{}
-	hnsV2 := hnsV2{}
-
-	testCreateEndpointLocal(t, hnsV1)
-	testCreateEndpointLocal(t, hnsV2)
-}
-func TestCreateEndpointRemotePA(t *testing.T) {
-	hnsV1 := hnsV1{}
-	hnsV2 := hnsV2{}
-
-	testCreateEndpointRemote(t, hnsV1, epPaAddress)
-	testCreateEndpointRemote(t, hnsV2, epPaAddress)
-}
-func TestCreateEndpointRemoteNoPA(t *testing.T) {
-	hnsV1 := hnsV1{}
-	hnsV2 := hnsV2{}
-
-	testCreateEndpointRemote(t, hnsV1, "")
-	testCreateEndpointRemote(t, hnsV2, "")
+func assertHCNDiff(x, y interface{}) string {
+	// We can ignore some fields since they are known only at creation time
+	var nx, ny interface{}
+	switch x.(type) {
+	case hcn.HostComputeEndpoint:
```

Comment:
We should also compare fields like IPAddress, MACAddress for endpoints, VIP, backends, load balancing type (DSR vs. non-DSR), and Ports for load balancers, and Network CIDR for networks, right?

Reply:
I think the assert function is comparing them already. It is comparing all fields, except the ones that are wrote there (Id, Name, Health etc.). I will take a closer look at this.